### ISSUE TITLE
Option to store the indexing instead of re-indexing every time

### ISSUE THREAD

**Issue by ecandir** [lunu-bounir/search-all-tabs#29]
I have a bunch of websites open and many pdfs open. Re-indexing these pages takes around 1 minute. Every time I search for something new, or a deviation of the search clause, it reindexes. This takes a lot of time. Maybe a setting can be added to store the index of already indexed pages. Otherwise, this is one of the best extensions:)

**Comment by freemedom** [comment]
I had the same problem. It took too long to index every query when I have a large number of tabs.

**Comment by freemedom** [comment]
I found a temporary solution,just need to keep the popup window is not closed.

**Comment by kapooky** [comment]
Yes, I want this feature as well. I regularly have 100's of tabs open, and use this extension to filter for expressions. I don't want want to wait for the database to be populated to use the search function.

**Comment by jacksongoode** [comment]
This is strange there's no incremental indexing based off of changes to the tab list.

**Comment by MisterAcoustic** [comment]
User tariqazmat101 has hundreds of tabs open. How quaint :). I have thousands of tabs open. This extension is no longer usable. I would be happy to let the indexing run to completion once, but when it runs on every search, there's no way I'm waiting - that's when I need the search, not the indexing. I presume that no work is being done on this, given the age of the issue - I just wanted to let someone know how bad it is for some users.

UPDATE: The answer from [freemedom](https://github.com/freemedom) above intrigued me, although I had no idea what they were talking about. I looked a little deeper, and found out what they meant. Here are some better instructions for anyone who got this far in trying to solve this problem:

Right-click on the toolbar search (magnifying glass) icon. You should see a dropdown containing a list. From the list, select "Options". Under the Options menu, find an entry called "Open Mode". This offers "Popup" and "Tab". The normal mode of operation is "Tab". Switch it to "Popup". This opens the search window as a tab, which can stay open as long as you like. It runs the indexing, and is ready to go once it has completed. You can then use the tab window to search, and the indexing will not run for each search attempt.

I still feel this should be addressed if possible, but I'm happier now that there is some sort of workaround.

**Comment by yshui** [comment]
I got persistent index kind of working... https://github.com/yshui/search-all-tabs-fork

It's for Firefox only. I made it in half a day so don't expect quality :sweat_smile: Also some of the options were removed because they are hard to support with a persistent index. If you want to try it you need to either load it via `about:debugging` or sign it yourself.
